lib/types/message.js

```diff
@@ -175,7 +175,14 @@ module.exports.encodeMessage = function(message, buffer) {
     // @todo:
     //  Array[Buffer] => multiple Data segments
     //  Array[Array] => multiple AMQPSequence
-    if (message.body instanceof Buffer) {
+    if (message.body instanceof DescribedType) {
+      var descriptor = message.body.descriptor;
+      if (!u.includes([0x75, 0x76, 0x77], descriptor)) {
+        throw new errors.MalformedPayloadError('Unknown DescribedType: ',
```

Comment:
just a little nitpick, but I think a more informative error might suit the user here. Like "Invalid described type for body", or probably something much better worded 😄